tempest/api/workloadmgr/barbican/test_barbican.py

```diff
@@ -3120,3 +3134,215 @@ def test_10_barbican(self):
+        try:
+            reporting.add_test_script(str(__name__) + \
+                    "_validate_scheduler_trust_with_scheduler_disabled")
+            global vm_id
+            global volume_id
+            global exception
+            global secret_uuid
+            if exception != "":
+                LOG.error("pre req failed")
+                raise Exception(str(exception))
+            LOG.debug("pre req completed")
+
+            # Create workload with scheduler disabled using CLI
+            workload_create = \
+                    command_argument_string.workload_create_with_encryption +\
+                " --instance instance-id=" + \
+                str(vm_id) + " --jobschedule enabled=False" + \
+                " --secret-uuid " + str(secret_uuid)
+            rc = cli_parser.cli_returncode(workload_create)
+            if rc != 0:
+                reporting.add_test_step(
+                    "Execute workload-create command with scheduler disable",
+                    tvaultconf.FAIL)
+                raise Exception(
+                    "Command workload create did not execute correctly")
+            else:
+                reporting.add_test_step(
+                    "Execute workload-create command with scheduler disable",
+                    tvaultconf.PASS)
+
+            time.sleep(10)
+            self.wid = query_data.get_workload_id_in_creation(tvaultconf.workload_name)
+            LOG.debug("Workload ID: " + str(self.wid))
+            if(self.wid is not None):
+                self.wait_for_workload_tobe_available(self.wid)
+                if(self.getWorkloadStatus(self.wid) == "available"):
+                    reporting.add_test_step(
+                        "Create workload with scheduler disable", tvaultconf.PASS)
+                else:
+                    reporting.add_test_step(
+                        "Create workload with scheduler disable", tvaultconf.FAIL)
+                    reporting.set_test_script_status(tvaultconf.FAIL)
+            else:
+                raise Exception("Create workload with scheduler disabled")
+
+            #Execute scheduler-trust-validate CLI command
+            self._execute_scheduler_trust_validate_cli()
+
+            #Fetch trust list from API
+            trust_list = self.get_trusts()
+
+            #Verify trust details returned
+            if not self.wlm_trust_valid and \
+                    not self.wlm_scheduler and \
+                    len(trust_list) > 0:
+                reporting.add_test_step("Verify trust", tvaultconf.PASS)
+            else:
+                reporting.add_test_step("Verify trust", tvaultconf.FAIL)
+                reporting.set_test_script_status(tvaultconf.FAIL)
+
+            #Delete the trust using API
+            if self.delete_trust(trust_list[0]['name']):
+                reporting.add_test_step("Delete trust", tvaultconf.PASS)
+            else:
+                raise Exception("Delete trust")
```

Comment:
Do you think we should add_test_step() to FAIL in else part - if delete_trust() fails before raising an exception.

Reply:
If delete_trust() fails, exception is handled in base.py itself and False is returned in that case. Accordingly this if block handles the failure.